routing/base/astar_algorithm.hpp

```diff
@@ -13,18 +13,43 @@
 namespace routing
 {
 
-template <typename TVertexType>
+template <typename TWeightType>
+constexpr TWeightType GetWeightZero();
+template <>
+constexpr double GetWeightZero<double>()
+{
+  return 0.0;
+}
+
+// Precision of comparison weights.
+template <typename TWeightType>
+constexpr TWeightType GetWeightEpsilon();
+template <>
+constexpr double GetWeightEpsilon<double>()
+{
+  return 1e-6;
+}
+
+template <typename TWeightType>
+constexpr TWeightType GetWeightMax();
+template <>
+constexpr double GetWeightMax<double>()
```

Comment:
No, you didn't get my idea. My idea is to modify RouteWeight in a following way:
```c++
class RouteWeight
{
public:
  ...

  static RouteWeight GetZero() { return RouteWeight(0.0); }
  static RouteWeight GetMax() { return RouteWeight(numeric_limits<double>::max()); }

  ...
};
```
and you may completely get rid of these GetAStarWeightZero and other functions. My proposal is to incorporate all these getters directly into the weight class.